main/src/main/scala/sbt/internal/server/BuildServerReporter.scala

```diff
@@ -69,11 +75,24 @@ sealed trait BuildServerReporter extends Reporter {
 
   override def comment(pos: XPosition, msg: String): Unit = underlying.comment(pos, msg)
 }
+/**
+
+An implementation of the BuildServerReporter for communicating with the Build Server.
+Sends diagnostic messages to the client, handling success and failure cases.
+@param buildTarget the identifier of the build target
+@param bspCompileState state representing the compilation of the Build Server Protocol
+@param converter a file converter for converting between VirtualFileRef and Path
+@param sourcePositionMapper a function to map an xsbti.Position from the generated file  to the original xsbti.Position
```

Comment:
a function to map an xsbti.Position from the generated file (the Scala file) to the source file of the generator (e.g. Twirl file)